documentation/decisions/adr/0004-api-framework.md

```diff
@@ -0,0 +1,111 @@
+# API Framework and Libraries
+
+- **Status:** Accepted on 2023-07-06 <!-- REQUIRED -->
+- **Last Modified:** 2023-07-07 <!-- REQUIRED -->
+- **Related Issue:** [#28](https://github.com/HHS/grants-api/issues/28) <!-- RECOMMENDED -->
+- **Deciders:** Lucas brown, Aaron Couch, Billy Daly, Sammy Steiner, Daphne Gold, Gina Carson, Sumi Thaiveettil, Eshter Oke <!-- REQUIRED -->
+
+## Context and Problem Statement
+
+This ADR is to decide what python framework to use for the backend API of grants.gov. Python was chosen as the language for the backend API in [ADR #3](https://github.com/HHS/grants-api/blob/main/documentation/decisions/adr/0003-api-language.md).
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- Is this framework well established with a broad community of developers
+- Does it have good documentation for developers to get up to speed quickly
+- Does this language have a track record for reliability and continued support
+- How familiar is Nava and HHS with the framework
+- How effectively can the chosen framework support the backend needs of grants.gov
+
+## Options Considered
+
+- Flask 
+- Flask + Connexion
+- Flask + APIFlask
+- FastAPI
+- Django
+
+## Decision Outcome <!-- REQUIRED -->
+
+Chosen option: Flask + APIFlask, because it is well established with a broad community of developers and provides good tooling to move quickly. It has great documentation to help folks contribute quickly. Additionally the Nava Flask template recently adopted it, so we can leverage the template to get going quickly with a well engineered solution.
```

Comment:
Thanks for adding this!! I was looking at APIFlask yesterday and seems like a sensible library with some cool features.

Reply:
Looks good to me! Thanks for the thoughtful consideration here @SammySteiner and others.